apps/server/src/modules/files-storage/client/s3-client.adapter.spec.ts

```diff
@@ -430,4 +430,166 @@ describe('S3ClientAdapter', () => {
 			await expect(service.copy(undefined)).rejects.toThrowError(InternalServerErrorException);
 		});
 	});
+
+	describe('head', () => {
+		const setup = () => {
+			const { pathToFile } = createParameter();
+
+			return { pathToFile };
+		};
+
+		it('should call send() of client with head object', async () => {
+			const { pathToFile } = setup();
+
+			await service.head(pathToFile);
+
+			expect(client.send).toBeCalledWith(
+				expect.objectContaining({
+					input: { Bucket: 'test-bucket', Key: pathToFile },
+				})
+			);
+		});
+	});
+
+	describe('list', () => {
+		const setup = () => {
+			const prefix = 'test/';
+
+			const keys = Array.from(Array(2500).keys()).map((n) => `KEY-${n}`);
+			const responseContents = keys.map((key) => {
+				return {
+					Key: `${prefix}${key}`,
+				};
+			});
+
+			return { prefix, keys, responseContents };
+		};
+
+		afterEach(() => {
+			client.send.mockClear();
+		});
+
+		it('should call send() of client', async () => {
+			const { prefix, keys, responseContents } = setup();
+
+			// @ts-expect-error should run into error
+			client.send.mockResolvedValue({
+				IsTruncated: false,
+				Contents: responseContents,
+			});
+
+			const resultKeys = await service.list(prefix);
+
+			expect(resultKeys).toEqual(keys);
+
+			expect(client.send).toBeCalledWith(
+				expect.objectContaining({
+					input: {
+						Bucket: 'test-bucket',
+						Prefix: prefix,
+						ContinuationToken: undefined,
+						MaxKeys: 1000,
+					},
+				})
+			);
+		});
+
+		it('should truncate result when max is given', async () => {
+			const { prefix, keys, responseContents } = setup();
+
+			// @ts-expect-error should run into error
```

Comment:
Why "should run into error"? Nothing runs into an error here...

Reply:
Typescript complains because the mock-function expects a parameter of type never. The comment was copied from the existing tests in that file

Reply:
Then "ignore parameter type of mock function" or something like that would be a good comment. "should run into an error" sounds like you want something (e.g. a subsequent function call) to run into an error.